public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,347 @@
+            getDetectors();
+          });
+      }
+    });
+  };
+
+  const getUnlinkConfirmModal = () => {
+    if (confirmModalState.isOpen) {
+      return (
+        <ConfirmUnlinkDetectorModal
+          detector={confirmModalState.affectedDetector}
+          onUnlinkDetector={onUnlinkDetector}
+          onHide={handleHideModal}
+          onConfirm={handleConfirmModal}
+          isListLoading={isLoading}
+        />
+      );
+    }
+  };
+
+  const handleHideModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isOpen: false,
+    });
+  };
+
+  const handleConfirmModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isRequestingToClose: true,
+    });
+  };
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  // TODO: this part is incomplete because it is pending on a different PR that will have all the associate existing changes
+  const onAssociateExistingDetector = async () => {
+    console.log('inside create anomaly detector');
+  };
+
+  const handleUnlinkDetectorAction = (detector: DetectorListItem) => {
+    setDetectorToUnlink(detector);
+    if (!isEmpty(detector)) {
+      setConfirmModalState({
+        isOpen: true,
+        action: ASSOCIATED_DETECTOR_ACTION.UNLINK,
+        isListLoading: false,
+        isRequestingToClose: false,
+        affectedDetector: detector,
+      });
+    } else {
+      core.notifications.toasts.addWarning(
+        'Make sure selected detector has not been deleted'
+      );
+    }
+  };
+
+  const columns = useMemo(
+    () => getColumns({ handleUnlinkDetectorAction }),
+    [handleUnlinkDetectorAction]
+  );
+
+  const renderEmptyMessage = () => {
+    if (isLoading) {
+      return 'Loading detectors...';
+    } else if (!isEmpty(selectedDetectors)) {
+      return (
+        <EmptyAssociatedDetectorFlyoutMessage
+          isFilterApplied={true}
+          embeddableTitle={embeddableTitle}
+        />
+      );
+    } else {
+      return (
+        <EmptyAssociatedDetectorFlyoutMessage
+          isFilterApplied={false}
+          embeddableTitle={embeddableTitle}
+        />
+      );
+    }
+  };
+
+  const tableProps = {
+    items: selectedDetectors,
+    columns,
+    search: {
+      box: {
+        disabled: selectedDetectors.length === 0,
+        incremental: true,
+        schema: true,
+      },
+    },
+    hasActions: true,
+    pagination: true,
+    sorting: true,
+    message: renderEmptyMessage(),
+  };
+  return (
+    <div className="associated-detectors">
+      <EuiFlyout ownFocus size="m" paddingSize="m" onClose={closeFlyout}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="s">
+            <h2 id="associated-detectors__title">
+              Associated anomaly detectors
+            </h2>
+          </EuiTitle>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody style={{ overflowY: 'auto' }}>
+          {getUnlinkConfirmModal()}
```

Comment:
nit - would it be cleaner to just instantiate the confirm modal jsx here instead of embedding within a function to fetch it?

Reply:
I can do this, just wanted to a shorted component here but no strong preference